gui/default/syncthing/folder/folderPickerModalController.js

```diff
@@ -0,0 +1,171 @@
+                return path.slice(0, -$scope.pathSeparator.length);
+            }
+            return path;
+        }
+
+        function expandTilde(path) {
+            if (path.startsWith('~')) {
+                return path.replace('~', $scope.$parent.system.tilde);
+            }
+            return path;
+        }
+
+        function splitPath(path) {
+            const trimmedPath = expandTilde(stripTrailingSeparator(path));
+            // Keep the leading separator if it exists
+            let parts = trimmedPath.startsWith($scope.pathSeparator)
+                ? [$scope.pathSeparator, ...trimmedPath.slice($scope.pathSeparator.length).split($scope.pathSeparator)]
+                : trimmedPath.split($scope.pathSeparator);
+
+            return parts.filter(Boolean);
+        }
+
+        function joinPath(...parts) {
+            // Don't add a second separator if the first part is already a separator
+            if (parts.length > 0 && parts[0] === $scope.pathSeparator) {
+                return $scope.pathSeparator + parts.slice(1).join($scope.pathSeparator);
+            }
+            return parts.join($scope.pathSeparator);
+        }
+
+        function formatDirectoryName(path) {
+            return splitPath(path).pop() || path
+        }
+
+        async function fetchSubdirectories(path) {
+            let res = await $http.get(urlbase + '/system/browse', {
+                params: {current: addTrailingSeparator(path)}
+            });
+            return res.data.map(dir => ({
+                title: formatDirectoryName(dir),
+                key: stripTrailingSeparator(dir),
+                folder: true,
+                lazy: true
+            }));
+        }
+
+        function clearTemporaryNodes(currentNode) {
+            // Delete temporary nodes, children first
+            $scope.tempNodes.sort((a, b) => b.getLevel() - a.getLevel());
+            $scope.tempNodes = $scope.tempNodes.filter(node => {
+                if (!node.children && node.key !== currentNode.key) {
+                    node.remove();
+                    return false;
+                }
+                return true;
+            });
+        }
+
+        async function findOrCreateNode(currentNode, key) {
+            if (currentNode.isLazy()) {
+                await currentNode.load();
+            }
+            if (!currentNode.children) {
+                currentNode.children = [];
+            }
+
+            let nextNode = currentNode.children.find(child => child.title === key);
+            // Prevent creating temp nodes at the root of the tree
+            if (!nextNode && currentNode !== $scope.tree.getRootNode()) {
+                nextNode = currentNode.addChildren({
+                    title: key,
+                    key: joinPath(currentNode.key, key),
+                    extraClasses: "fancytree-temp",
+                    folder: true,
+                    tooltip: $translate.instant('Folder will be created')
```

Comment:
Just wondering if it would be possible to use a Bootstrap tooltip here. Normally, it is done by attributes in the HTML, e.g.

```
<span tooltip data-original-title="{{'Folder will be created' | translate}}">
```